test/Release/Gateway/InMemoryReleaseGateway.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * @author: Ryan Wood <ryanw@clearbooks.co.uk>
+ * @created: 15/07/15
+ */
+
+namespace Clearbooks\Labs\Release\Gateway;
+
+use Clearbooks\Labs\Release\Release;
+
+class InMemoryReleaseGateway implements ReleaseGateway
```

Comment:
I would avoid using Fakes, they have a tendency to get unwieldy. You should attempt to use Dummies/Stubs/Spies/True Mocks instead.

The reason being is that Fakes tend to look very much like the thing they are Faking, and need to replicate all that business logic baggage. And for every new feature you add to a production version, you have to update all your fakes in kind.